devices/xiaomi.js

```diff
@@ -1875,22 +1875,38 @@ module.exports = [
         vendor: 'Xiaomi',
         whiteLabel: [{vendor: 'Xiaomi', model: 'CM-M01'}],
         description: 'Aqara curtain driver E1',
-        fromZigbee: [fz.battery, fz.xiaomi_curtain_position_tilt, fz.aqara_opple, fz.power_source],
-        toZigbee: [tz.xiaomi_curtain_position_state, tz.ZNCLBL01LM_battery_voltage, tz.ZNCLBL01LM_hooks_state,
-            tz.power_source, tz.battery_percentage_remaining],
-        exposes: [e.cover_position().setAccess('state', ea.ALL), e.battery().withAccess(ea.STATE_GET),
+        fromZigbee: [
+            fz.battery,
+            fz.xiaomi_curtain_position_tilt,
+            fz.aqara_opple,
+            fz.power_source,
+        ],
+        toZigbee: [
+            tz.xiaomi_curtain_position_state,
+            tz.xiaomi_curtain_battery_voltage,
+            tz.ZNCLBL01LM_hooks_action,
+            tz.ZNCLBL01LM_hooks_state,
+            tz.ZNCLBL01LM_hand_open,
+            tz.ZNCLBL01LM_limits_calibration,
+            tz.power_source,
+            tz.battery_percentage_remaining,
+        ],
+        exposes: [
+            e.cover_position().setAccess('state', ea.ALL),
+            exposes.binary('hand_open', ea.ALL, true, false).withDescription('Pulling curtains by hand starts the motor'),
+            exposes.enum('limits_calibration', ea.SET, ['start', 'end', 'reset']).withDescription('Calibrate the position limits'),
+            e.battery().withAccess(ea.STATE_GET),
             e.battery_voltage().withAccess(ea.STATE_GET),
             e.device_temperature(),
             e.action(['manual_open', 'manual_close']),
-            exposes.enum('motor_state', ea.STATE, ['stopped', 'opening', 'closing', 'pause'])
-                .withDescription('Motor state'),
-            exposes.binary('running', ea.STATE, true, false)
-                .withDescription('Whether the motor is moving or not'),
-            exposes.enum('hooks_state', ea.STATE_GET, ['unlocked', 'locked', 'locking', 'unlocking'])
-                .withDescription('Hooks state'),
+            exposes.enum('motor_state', ea.STATE, ['stopped', 'opening', 'closing', 'pause']).withDescription('Motor state'),
+            exposes.binary('running', ea.STATE, true, false).withDescription('Whether the motor is moving or not'),
+            exposes.enum('hooks_action', ea.SET, ['unlock', 'lock']).withDescription('Hooks action'),
```

Comment:
This should be called `action_hooks` (which will make the value get cleared when the HA integration is enabled).

Reply:
Wouldn't that collide somehow with the existing read-only field also called `action`?

![Skärmavbild 2022-12-09 kl  01 51 18](https://user-images.githubusercontent.com/4631744/206597952-ad353430-adb1-430a-8eee-a6930c85ef1c.png)

Reply:
No, since `action` is not a state (something that has to be persisted).